pretyping/coercion.ml

```diff
@@ -552,45 +550,46 @@ let rec inh_conv_coerce_to_fail ?loc env evd ?(flags=default_flags_of env) rigid
             | na -> na) name in
           let open Context.Rel.Declaration in
           let env1 = push_rel (LocalAssum (name,u1)) env in
-          let (evd', v1, trace1) =
-            inh_conv_coerce_to_fail ?loc env1 evd rigidonly
+          let (sigma, v1, trace1) =
+            inh_conv_coerce_to_fail ?loc env1 sigma rigidonly
               (mkRel 1) (lift 1 u1) (lift 1 t1) in
-          let v2 = beta_applist evd' (lift 1 v,[v1]) in
-          let t2 = Retyping.get_type_of env1 evd' v2 in
-          let (evd'',v2',trace2) = inh_conv_coerce_to_fail ?loc env1 evd' rigidonly v2 t2 u2 in
+          let v2 = beta_applist sigma (lift 1 v,[v1]) in
+          let t2 = Retyping.get_type_of env1 sigma v2 in
+          let (sigma,v2',trace2) = inh_conv_coerce_to_fail ?loc env1 sigma rigidonly v2 t2 u2 in
```

Comment:
BTW I removed all `evd` to `sigma` and removed `'` to make sure we didn't miss a shadowing somewhere.